Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemsController.cs

```diff
@@ -737,6 +806,16 @@ private void AddTestsToProblem(Problem problem, HttpPostedFileBase testArchive)
             }
         }
 
+        private void ValidateUploadedFile(string propertyName, HttpPostedFileBase additionalFiles)
+        {
+            var extension = additionalFiles.FileName.Substring(additionalFiles.FileName.Length - 4, 4);
```

Comment:
Better find or implement some method for getting the extension. The class `Path` must have one.